src/index.js

```diff
@@ -303,18 +302,25 @@ class Dayjs {
     })
   }
 
+  utcOffset() {
+    // Because a bug at FF24, we're rounding the timezone offset around 15 minutes
+    // https://github.com/moment/moment/pull/1871
+    return -Math.round(this.$d.getTimezoneOffset() / 15) * 15
+  }
+
   diff(input, units, float) {
     const unit = Utils.prettyUnit(units)
     const that = dayjs(input)
+    const zoneDelta = (that.utcOffset() - this.utcOffset()) * C.MILLISECONDS_A_MINUTE
     const diff = this - that
     let result = Utils.monthDiff(this, that)
 
     result = {
       [C.Y]: result / 12,
       [C.M]: result,
       [C.Q]: result / 3,
-      [C.W]: diff / C.MILLISECONDS_A_WEEK,
-      [C.D]: diff / C.MILLISECONDS_A_DAY,
+      [C.W]: (diff - zoneDelta) / C.MILLISECONDS_A_WEEK,
+      [C.D]: (diff - zoneDelta) / C.MILLISECONDS_A_DAY,
```

Comment:
Why we just need zoneDelta only in week and day? 
Could this change added to  `diff` in line 315?
```
const diff = this - that - zoneDelta
```

Reply:
I'm not sure 😅 but substracting the zoneDelta to other units than `week` and `day`, makes the tests broke.

Maybe because units under `day` do not consider daysaving?